docs/user-guide/petset.md

```diff
@@ -0,0 +1,370 @@
+    | mysql-0 | <-> | mysql-1 |
+      [pd 0]          [pd 1]
+```
+
+The co-ordinated deployment of clustered applications is notoriously hard. They require stronger notions of identity and membership, which they use in opaque internal protocols, and are especially prone to race conditions and deadlock. Traditionally administrators have deployed these applications by leveraging nodes as stable, long-lived entities with persistent storage and static ips.
+
+The goal of Pet Set is to decouple this dependency by assigning identities to individual instances of an application that are not anchored to the underlying physical infrastructure. For the rest of this document we will refer to these entities as "Pets". Our use of this term is predated by the "Pets vs Cattle" analogy. These are the properties of a Pet:
+
+* a stable hostname, available in DNS
+* an ordinal index
+* stable storage: linked to the ordinal & hostname
+* discovery of peers for quorum
+* startup/teardown ordering
+
+## Alpha limitations
+
+Before you start deploying applications as Pet Sets, there are a few limitations you should understand.
+
+* Pet Set is an *alpha* resource, not available in any Kubernetes release prior to 1.3.
+* As with all alpha/beta resources, it can be disable through the `--runtime-config` option passed to the apiserver, and in fact most likely will be disabled on hosted offerings of Kubernetes.
+* The only updatable field on a Pet Set is `replicas`
+* The storage for a given pet is provisioned by a [dynamic storage provisioner](https://github.com/kubernetes/kubernetes/blob/release-1.3/examples/experimental/persistent-volume-provisioning/README.md) based on the requested `storage class`. Note that dynamic volume provisioning is also currently in alpha.
+* Deleting the Pet Set  *will not*  delete all pets. You will either have to manually scale it down to 0 pets first, or delete the pets yourself.
+* Deleting and/or scaling a Pet Set down will *not* delete the volumes associated with the Pet Set. This is done to ensure safety first, your data is more valuable than an auto purge of all related Pet Set resources. Deleting the Persistent Volume Claims will result in a deletion of the associated volumes.
+* All Pet Sets currently require a "governing service", or a Service responsible for the network identity of the pets. The user is responsible for this Service.
+* Updating an existing Pet Set is currently a manual process, meaning you either need to deploy a new Pet Set with the new image version, or orphan Pets one by one, update their image, and join them back to the cluster.
+
+
+## Properties of a Pet
+
+The following examples demonstrate the properties of Pets.
+
+### Identity
+
+A minimal Pet Set might look like:
+
+{% include code.html language="yaml" file="petset.yaml" ghlink="/docs/user-guide/petset.yaml" %}
+
+For completeness it includes the `Service` definition. Lets create the Pet Set:
+
+```shell
+$ kubectl create -f petset.yaml
```

Comment:
@bprashanth You may want to give the full path from kubernetes root for the petset.yaml file.